core/src/main/java/org/apache/cxf/common/util/Compiler.java

```diff
@@ -85,6 +85,31 @@ public void setClassPath(String s) {
         classPath = StringUtils.isEmpty(s) ? null : s;
     }
 
+    // https://issues.apache.org/jira/browse/CXF-8049
+    private String getSystemClassPath() {
+        String javaClasspath = SystemPropertyAction.getProperty("java.class.path");
+
+        if (!StringUtils.isEmpty(javaClasspath)) {
+            List<String> correctedEntries = new ArrayList<>();
+
+            String[] toks = javaClasspath.split(File.pathSeparator);
+            
+            for (String tok: toks) {
+                // if any classpath entry contains a whitespace char, 
+                // enclose the entry in double quotes
+                if (tok.matches(".*\\s+.*")) {
+                    correctedEntries.add("\"" + tok + "\"");
+                } else {
+                    correctedEntries.add(tok);
+                }
+            }
+
+            return String.join(File.pathSeparator, correctedEntries);
+        }
+
+        return "";
```

Comment:
It could just return javaClasspath here

Reply:
+1
@coheigea , sorry I missed your comment and merged the PR, I will make another commit to address accordingly.

Cheers
Freeman